ocaml/util/xapi_version.ml

```diff
@@ -9,5 +9,14 @@
 let git_id = ""
 let hostname = "localhost"
 let date = Build_info.date
-let xapi_version_major = 1
-let xapi_version_minor = 20
+
+let (xapi_version_major, xapi_version_minor) =
+  (* v will be interpolated when:
+     - creating an RPM (`make version`)
+     - compiling with opam (`dune subst`)
+  *)
+  let v = "%%VERSION_NUM%%" in
+  try
+    Scanf.sscanf v "%d.%d.%s" (fun maj min _rest -> (maj, min))
+  with _ ->
+    (0, 0)
```

Comment:
I would rather makes this utterly fail. We don't want to release a version that advertises itself as 0.0.

I would also like to expose the hash commit (now part of `_rest`), if it's available. I guess a `xapi_version_revision = Some rev` would do, along with modifying the code that exposes the version.

Reply:
if we allow this to fail, then you can't compile xapi without interpolating the version (i think this is because of the autogenerated code)

but interpolating the version means you have to carry around version changes in the working directory which is a pain

the  rev is available (if you use dune subst) it looks like this = 1.244.0-1-g0303cdaf1. i think we should leave implementing this until we know exactly what we need for koji